great_expectations/data_context/data_context/base_data_context.py

```diff
@@ -2250,7 +2255,16 @@ def save_profiler(
         else:
             key = ConfigurationIdentifier(configuration_key=name)
 
-        self.profiler_store.set(key=key, value=profiler.config)
+        response = self.profiler_store.set(key=key, value=profiler.config)
+        if isinstance(response, GeCloudResourceRef):
+            ge_cloud_id = response.ge_cloud_id
+
+        # If an id is present, we want to prioritize that as our key for object retrieval
+        if ge_cloud_id:
+            name = None
```

Comment:
Feels a little weird but we do an XOR with id and name in `get_profiler`.

Should we allow both but prioritize the id?

Reply:
I agree.
Don't know all the ins & outs of this but, this feels like a decision we should delegate to `get_profiler()`

Reply:
So `get_profiler` is used as a helper here but is also its own stand alone method for object retrieval. If a user tries to get their profiler using both name and id, what is the expected behavior? Should we error or have some order of precedence? We're currently using the latter pattern and in order to get these changes to work with that, we need to ensure that we're getting rid of any conflicts here before invoking the `get_profiler()` method.